lib/mail-archive-helper.ts

```diff
@@ -199,7 +199,43 @@ export class MailArchiveGitHelper {
              */
             this.state.latestRevision =
                 "3b38d8d206c64bf3dc873ba8ae9dbd48ed43f612";
+        } else if (await revParse(this.state.latestRevision,
+                                  this.mailArchiveGitDir) === undefined) {
+            const publicInboxGitDir = process.env.PUBLIC_INBOX_DIR;
+            if (publicInboxGitDir === undefined) {
+                throw new Error(`Commit ${this.state.latestRevision
+                                } not found; need PUBLIC_INBOX_DIR`);
+            }
+            let commitDiff =
+                await git(["show", this.state.latestRevision, "--"],
+                          { workDir: publicInboxGitDir });
+            if (!commitDiff) {
+                throw new Error(`Could not find ${this.state.latestRevision
+                                } in ${publicInboxGitDir}`);
+            }
+            const match = commitDiff.match(/\n\+(Message-ID: [^\n]*)/);
+            if (!match) {
+                throw new Error(`Could not find Message-ID in ${commitDiff}`);
+            }
+            let commit = "HEAD";
+            for (;;) {
+                commit = await git(["log", "-1", "--format=%H", `-S${match[1]}`,
+                                    commit, "--"],
+                                   { workDir: this.mailArchiveGitDir});
+                if (!commit) {
+                    throw new Error(`Could not find ${this.state.latestRevision
+                                }'s equivalent in ${this.mailArchiveGitDir}`);
+                }
+                commitDiff = await git(["show", commit, "--"],
+                                       {workDir: this.mailArchiveGitDir});
+                if (commitDiff.indexOf(`\n+${match[1]}\n`) >= 0) {
```

Comment:
I am not advocating for the removal but is the `+` really needed for the regex or was it just left in when the replacement mode of updates was found?

Reply:
It is actually necessary: in the public-inbox.org repository, every mail was added as its own file, but in the lore.kernel.org one, every new mail replaces the previously-new mail in the file `m`. Therefore, if we look only for the `Message-ID: <id>`, we may very well find a commit that _removes_ those contents, i.e. a commit that corresponds to the child of the commit we are looking for.

Reply:
So the `\n+` should be `\n\+`, to actually include the literal `+` or the match at line 216 could have included the `+` - ie `\n(\+Message-ID...)`.

Reply:
Oy. Oy, oy, oy. I got it right in line 216 and not here.

Good eyes! 👓

Reply:
Fixed ;-)

Reply:
Actually bad eyes. Sorry about this.  The second check is using `indexOf` so it is not a regex.  I was too focused on the search string and missed the method being used.  Realized my error while you were making the update.

Reply:
Oh. Okay, I reverted this. I clearly need more sleep.